core/node/proof_data_handler/src/tests.rs

```diff
@@ -0,0 +1,247 @@
+
+    let db_conn_pool = ConnectionPool::test_pool().await;
+    mock_tee_batch_status(db_conn_pool.clone(), batch_number, &object_path).await;
+
+    // test the /tee_proof_inputs endpoint; it should return the batch from the object store
+
+    let app = create_proof_processing_router(
+        blob_store,
+        db_conn_pool,
+        ProofDataHandlerConfig {
+            http_port: 1337,
+            proof_generation_timeout_in_secs: 10,
+        },
+        L1BatchCommitmentMode::Rollup,
+    );
+    let req_body = Body::from(serde_json::to_vec(&json!({})).unwrap());
+    let response = app
+        .oneshot(
+            Request::builder()
+                .method(Method::POST)
+                .uri("/tee_proof_inputs")
+                .header(http::header::CONTENT_TYPE, mime::APPLICATION_JSON.as_ref())
+                .body(req_body)
+                .unwrap(),
+        )
+        .await
+        .unwrap();
+
+    assert_eq!(response.status(), StatusCode::OK);
+
+    let body = hyper::body::to_bytes(response.into_body()).await.unwrap();
+    let json: serde_json::Value = serde_json::from_slice(&body).unwrap();
+    let json = json
+        .get("Success")
+        .expect("Unexpected response format")
+        .clone();
+    let deserialized: TeeVerifierInput = serde_json::from_value(json).unwrap();
+
+    assert_eq!(tvi, deserialized);
+}
+
+// Test /submit_tee_proof endpoint using a mocked TEE proof and verify response and db state
+#[tokio::test]
+async fn submit_tee_proof() {
+    let blob_store = ObjectStoreFactory::mock().create_store().await;
+    let db_conn_pool = ConnectionPool::test_pool().await;
+    let object_path = "mocked_object_path";
+    let batch_number = L1BatchNumber::from(1);
+
+    mock_tee_batch_status(db_conn_pool.clone(), batch_number, object_path).await;
+
+    // send a request to the /submit_tee_proof endpoint, using a mocked TEE proof
+
+    let tee_proof_request_str = r#"{
+        "Proof": {
+            "signature": [ 0, 1, 2, 3, 4 ],
+            "pubkey": [ 5, 6, 7, 8, 9 ],
+            "proof": [ 10, 11, 12, 13, 14 ]
+        }
+    }"#;
+    let tee_proof_request =
+        serde_json::from_str::<SubmitTeeProofRequest>(tee_proof_request_str).unwrap();
+    let uri = format!("/submit_tee_proof/{}", batch_number.0);
+    let app = create_proof_processing_router(
+        blob_store,
+        db_conn_pool.clone(),
+        ProofDataHandlerConfig {
+            http_port: 1337,
+            proof_generation_timeout_in_secs: 10,
+        },
+        L1BatchCommitmentMode::Rollup,
+    );
+
+    // this should fail because we haven't saved the attestation for the pubkey yet
+
+    let response = send_submit_tee_proof_request(&app, &uri, &tee_proof_request).await;
+    assert_eq!(response.status(), StatusCode::BAD_GATEWAY);
+
+    // save the attestation for the pubkey
+
+    if let SubmitTeeProofRequest::Proof(ref proof) = tee_proof_request {
+        let valid_until = Utc::now() + Duration::days(42);
```

Comment:
IMHO, we agreed on not getting the collateral in the SGX worker and therefore `valid_until` is unknown and not needed.

Reply:
[Fixed](https://github.com/matter-labs/zksync-era/pull/1993/commits/c0414f269f79cc8b67931b4d79c9b3b40820187f) ✅